sycl/unittests/windows/dllmain.cpp

```diff
@@ -0,0 +1,65 @@
+//==----- dllmain.cpp --- verify behaviour of lib on process termination ---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+/*
+ * This test calls DllMain on Windows. This means, the process performs actions
+ * which are required for library unload. That said, the test requires to be a
+ * distinct binary executable.
+ */
+
+#include <CL/sycl.hpp>
+#include <helpers/CommonRedefinitions.hpp>
+#include <helpers/PiImage.hpp>
+#include <helpers/PiMock.hpp>
+
+#include <gtest/gtest.h>
+
+#ifdef _WIN32
+#include <windows.h>
+
+extern "C" BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason,
+                               LPVOID lpReserved);
+
+static std::atomic<int> TearDownCalls{0};
+
+pi_result redefinedTearDown(void *PluginParameter) {
+  fprintf(stderr, "intercepted tear down\n");
```

Comment:
Could you please clarify, why `fprintf` is required here?

Reply:
I only got used to `fprintf(stderr, ...)` more than to `printf()`.
I prefer `printf` and `fprintf` over `std::cout` or `std::cerr` due the latter two tend to flush each call of `operator<<`.

Reply:
Got it. I guess committee has a holiday present for you then: http://open-std.org/JTC1/SC22/WG21/docs/papers/2021/p2093r9.html

Reply:
> I prefer printf and fprintf over std::cout or std::cerr due the latter two tend to flush each call of operator<<.

I thought only std::cerr flush operator<<. Why don't you use std::clog? Are you aware of it?